composer/checkpoint/load.py

```diff
@@ -539,10 +539,12 @@ def torch_set_model_state_dict(
                     textwrap.dedent(
                         'PyTorch DTensor broke backwards compatibility in older checkpoints '
                         'with ShardedTensor, which is now deprecated. To load old checkpoints, '
-                        'either downgrade to PyTorch <2.3.0 or explicitly pass process groups '
-                        'in the Trainer constructor via '
-                        "`parallelism_config = {'fsdp': {'process_group': 'mod1'}}`. We can "
-                        'provide assistance at https://github.com/mosaicml/composer/issues.',
+                        'either downgrade to PyTorch <2.3.0 or define process groups in the '
+                        'Trainer constructor via the `data_parallel_shard_degree` or '
+                        '`data_parallel_replicate_degree` arguments to `parallelism_config` via'
+                        'the "fsdp" key, as '
+                        '`parallelism_config = {"fsdp": {"data_parallel_shard_degree": 8}}`, '
+                        'for example. We can provide assistance at https://github.com/mosaicml/composer/issues.',
```

Comment:
This doesn't work -- it would create a device mesh and not sharded tensor with new API

Reply:
resolved offline